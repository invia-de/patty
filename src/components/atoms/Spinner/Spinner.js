import React from 'react';
import PropTypes from 'prop-types';

import styles from './spinner.module.scss';
import cx from '../../../utils/classnames';

/**
 * @author [Roman Semko](mailto:roman.semko-extern@invia.de)
 */
export default function Spinner({ className }) {
  return (
    <div className={cx(styles.spinner, className)}>
      <div className={styles.bounce1} />
      <div className={styles.bounce2} />
      <div className={styles.bounce3} />
    </div>
  );
}
Spinner.propTypes = {
  /** additional classNames you want to add */
  className: PropTypes.string
};
