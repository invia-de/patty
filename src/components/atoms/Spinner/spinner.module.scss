@import 'theme.scss';

$spinner-color: $blue !default;

.spinner {
  width: 84px;
  display: flex;
  justify-content: space-between;
}

.spinner > div {
  width: 24px;
  height: 24px;
  background-color: $spinner-color;

  border-radius: 100%;
  display: inline-block;
  animation: bouncedelay 1.4s infinite ease-in-out;
  /* Prevent first frame from flickering when animation starts */
  animation-fill-mode: both;
}

.spinner .bounce1 {
  animation-delay: -0.32s;
}

.spinner .bounce2 {
  animation-delay: -0.16s;
}

@keyframes bouncedelay {
  0%,
  80%,
  100% {
    transform: scale(0);
  }
  40% {
    transform: scale(1);
  }
}
