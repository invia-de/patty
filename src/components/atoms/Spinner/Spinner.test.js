import React from 'react';
import Spinner from './Spinner';
import { render } from '@testing-library/react';

test('Spinner renders', () => {
  const { container } = render(<Spinner />);
  expect(container.firstChild).toMatchSnapshot();
});
