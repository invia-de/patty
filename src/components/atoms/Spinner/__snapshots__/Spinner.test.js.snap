// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders 1`] = `
<div
  class="spinner"
>
  <div
    class="bounce1"
  />
  <div
    class="bounce2"
  />
  <div
    class="bounce3"
  />
</div>
`;
